peps/pep-0728.rst

```diff
@@ -143,96 +143,129 @@ This proposal aims to support a similar feature without introducing general
 intersection of types or syntax changes, offering a natural extension to the
 existing type consistency rules.
 
-We propose that we give the dunder attribute ``__extra__`` a special meaning:
-When it is defined on a TypedDict type, extra items are allowed, and their types
-should be compatible with the value type of ``__extra__``. Different from index
-signatures, the types of known items do not need to be consistent with the value
-type of ``__extra__``.
+We propose that we add an argument ``closed`` to TypedDict. Similar to
+``total``, only a literal ``True`` or ``False`` value is allowed. When
+``closed=True`` is used in the TypedDict type definition, we give the dunder
+attribute ``__extra_items__`` a special meaning: extra items are allowed, and
+their types should be compatible with the value type of ``__extra_items__``.
```

Comment:
I feel the case where ``__extra_items__`` is not set should be made more prominent; it's currently a bit buried further down.

```suggestion
their types should be compatible with the value type of ``__extra_items__``.
if ``closed=True`` is set, but there is no ``__extra_items__`` key, the TypedDict
is treated as if it contained an item ``__extra_items__: Never``.
```

Reply:
This is still buried a long way down. It belongs in the first few paragraphs of the spec.